File: src/Services/Roslyn/Markify.Services.Roslyn.Fixtures/Projects/Source/Class/ClassEvents.cs
using System;

public abstract class AbstractFooType
{
    public abstract event EventHandler AbstractEvent;
}

public class FooType : AbstractFooType
{
    event EventHandler PrivateEvent;

    internal event EventHandler InternalEvent;

    protected internal event EventHandler ProtectedInternalEvent;

    public static event EventHandler StaticEvent;

    public virtual event EventHandler VirtualEvent;

    public sealed override event EventHandler AbstractEvent;

    public event EventHandler<EventArgs> GenericEvent;

    public event EventHandler ExplicitEvent
    {
        add { }
        remove { }
    }
}

File: src/Services/Roslyn/Markify.Services.Roslyn.Fixtures/Projects/Source/Class/ClassEvents.vb
Imports System

Public Class FooType 
    Event PrivateEvent As EventHandler

    Friend Event InternalEvent As EventHandler

    Protected Friend Event ProtectedInternalEvent As EventHandler

    Public Shared Event StaticEvent As EventHandler

    Public Event GenericEvent As EventHandler(Of EventArgs)

    Public Custom Event ExplicitEvent As EventHandler
        AddHandler (ByVal value As EventHandler)
        End AddHandler
        RemoveHandler(ByVal value As EventHandler)
        End RemoveHandler
		RaiseEvent(ByVal sender As Object, ByVal e As EventArgs)
		End RaiseEvent
    End Event
End Class